File: packages/rook_sdk_health_connect/lib/src/data/mapper/health_data_type_mappers.dart
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk_health_connect/src/domain/enums/hc_health_data_type.dart';

extension HealthDataTypeMappers on HCHealthDataType {
  HealthDataTypeProto toProto() {
    return switch (this) {
      HCHealthDataType.sleepSummary => HealthDataTypeProto.SLEEP_SUMMARY,
      HCHealthDataType.physicalSummary => HealthDataTypeProto.PHYSICAL_SUMMARY,
      HCHealthDataType.bodySummary => HealthDataTypeProto.BODY_SUMMARY,
      HCHealthDataType.physicalEvent => HealthDataTypeProto.PHYSICAL_EVENT,
      HCHealthDataType.bloodGlucoseBodyEvent =>
        HealthDataTypeProto.BLOOD_GLUCOSE_BODY_EVENT,
      HCHealthDataType.bloodPressureBodyEvent =>
        HealthDataTypeProto.BLOOD_PRESSURE_BODY_EVENT,
      HCHealthDataType.bodyMetricsEvent =>
        HealthDataTypeProto.BODY_METRICS_EVENT,
      HCHealthDataType.heartRateBodyEvent =>
        HealthDataTypeProto.HEART_RATE_BODY_EVENT,
      HCHealthDataType.heartRatePhysicalEvent =>
        HealthDataTypeProto.HEART_RATE_PHYSICAL_EVENT,
      HCHealthDataType.hydrationBodyEvent =>
        HealthDataTypeProto.HYDRATION_BODY_EVENT,
      HCHealthDataType.nutritionBodyEvent =>
        HealthDataTypeProto.NUTRITION_BODY_EVENT,
      HCHealthDataType.oxygenationBodyEvent =>
        HealthDataTypeProto.OXYGENATION_BODY_EVENT,
      HCHealthDataType.oxygenationPhysicalEvent =>
        HealthDataTypeProto.OXYGENATION_PHYSICAL_EVENT,
      HCHealthDataType.temperatureBodyEvent =>
        HealthDataTypeProto.TEMPERATURE_BODY_EVENT,
    };
  }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/HealthDataTypeMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.HealthDataType
import com.rookmotion.rook_sdk_health_connect.exception.UnknownDataTypeException
import com.rookmotion.rook_sdk_health_connect.data.proto.HealthDataTypeProto

fun HealthDataTypeProto.toDomain(): HealthDataType {
    return when (this) {
        HealthDataTypeProto.SLEEP_SUMMARY -> HealthDataType.SLEEP_SUMMARY
        HealthDataTypeProto.PHYSICAL_SUMMARY -> HealthDataType.PHYSICAL_SUMMARY
        HealthDataTypeProto.BODY_SUMMARY -> HealthDataType.BODY_SUMMARY
        HealthDataTypeProto.PHYSICAL_EVENT -> HealthDataType.PHYSICAL_EVENT
        HealthDataTypeProto.BLOOD_GLUCOSE_BODY_EVENT -> HealthDataType.BLOOD_GLUCOSE_BODY_EVENT
        HealthDataTypeProto.BLOOD_PRESSURE_BODY_EVENT -> HealthDataType.BLOOD_PRESSURE_BODY_EVENT
        HealthDataTypeProto.BODY_METRICS_EVENT -> HealthDataType.BODY_METRICS_EVENT
        HealthDataTypeProto.HEART_RATE_BODY_EVENT -> HealthDataType.HEART_RATE_BODY_EVENT
        HealthDataTypeProto.HEART_RATE_PHYSICAL_EVENT -> HealthDataType.HEART_RATE_PHYSICAL_EVENT
        HealthDataTypeProto.HYDRATION_BODY_EVENT -> HealthDataType.HYDRATION_BODY_EVENT
        HealthDataTypeProto.NUTRITION_BODY_EVENT -> HealthDataType.NUTRITION_BODY_EVENT
        HealthDataTypeProto.OXYGENATION_BODY_EVENT -> HealthDataType.OXYGENATION_BODY_EVENT
        HealthDataTypeProto.OXYGENATION_PHYSICAL_EVENT -> HealthDataType.OXYGENATION_PHYSICAL_EVENT
        HealthDataTypeProto.TEMPERATURE_BODY_EVENT -> HealthDataType.TEMPERATURE_BODY_EVENT
        else -> throw UnknownDataTypeException()
    }
}
